core/domain/image_services.py

```diff
@@ -44,44 +49,35 @@ def get_image_dimensions(file_content):
     Returns:
         tuple(int). Returns height and width of the image.
     """
-    img = Image.open(io.BytesIO(file_content))
-    width, height = img.size
-    return height, width
+    image = Image.open(io.BytesIO(file_content))
+    return _get_pil_image_dimensions(image)
 
 
 def compress_image(image_content, scaling_factor):
     """Compresses the image by resizing the image with the scaling factor.
 
-    Note that if the image's dimensions, after the scaling factor is applied,
-    exceed 4000 then the scaling factor will be recomputed and applied such that
-    the larger dimension of the image does not exceed 4000 after resizing. This
-    is due to an implementation limitation. See https://goo.gl/TJCbmE for
-    context.
-
     Args:
         image_content: str. Content of the file to be compressed.
         scaling_factor: float. The number by which the dimensions of the image
-            will be scaled. This is expected to be greater than zero.
+            will be scaled. This is expected to be in the interval (0, 1].
 
     Returns:
         str. Returns the content of the compressed image.
     """
-    if not constants.DEV_MODE:
-        height, width = get_image_dimensions(image_content)
-        new_width = int(width * scaling_factor)
-        new_height = int(height * scaling_factor)
-        if (new_width > MAX_RESIZE_DIMENSION_PX
-                or new_height > MAX_RESIZE_DIMENSION_PX):
-            # Recompute the scaling factor such that the larger dimension does
-            # not exceed 4000 when scaled.
-            new_scaling_factor = (
-                python_utils.divide(
-                    MAX_RESIZE_DIMENSION_PX, float(max(width, height))))
-            new_width = int(width * new_scaling_factor)
-            new_height = int(height * new_scaling_factor)
-        return images.resize(
-            image_data=image_content,
-            width=min(new_width, MAX_RESIZE_DIMENSION_PX),
-            height=min(new_height, MAX_RESIZE_DIMENSION_PX))
-    else:
-        return image_content
+    if scaling_factor > 1 or scaling_factor <= 0:
+        raise ValueError('Scaling factor should be in the interval (0, 1].')
+    image = Image.open(io.BytesIO(image_content))
+
+    image_format = image.format
+    height, width = _get_pil_image_dimensions(image)
+    new_width = int(width * scaling_factor)
+    new_height = int(height * scaling_factor)
+    new_image_dimensions = (new_width, new_height)
```

Comment:
On line 72 I see (h, w) whereas here it's (w, h) it makes me wonder that we are doing something wrong in first glance, should we use the same pattern in both the places? (not sure!)

Reply:
It's OK to swap it as long as you're careful to swap everything else too and there's no other good reason for keeping (h, w). I think it's optional though.

Reply:
It can also be done in a separate PR, if you prefer.